### PR TITLE
Configure TKG_HTTP_PROXY_ENABLED variable if proxy is set during upgrade

### LINKED ISSUE #1468: Cluster upgrade fails for clusters for proxied environment
**Bug description**

If the proxy is enabled on the cluster, doing an upgrade fails unless the user specifies the `TKG_HTTP_PROXY_ENABLED*` environment variable manually.

Regression might be caused by this change to add validation around this config variable: https://github.com/vmware-tanzu/tanzu-framework/pull/793

During the upgrade process, `TKG_HTTP_PROXY_ENABLED` is not getting set when the proxy configuration is set.

https://github.com/vmware-tanzu/tanzu-framework/blob/92e10826f34f455f96b98fb56481e92078fe5278/pkg/v1/tkg/client/upgrade_addon.go#L328-L338

Error message:
```
Applying patch to resource wc-05 of type *v1beta1.Cluster ...
Error: failed to update additional addon components: unable to get cluster configuration: workload cluster configuration validation failed: cannot get TKG_HTTP_PROXY_ENABLED: Failed to get value for variable "TKG_HTTP_PROXY_ENABLED". Please set the variable value using os env variables or using the config file
```

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**
Upgrade to be successful.

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by anujc25**
- As part of PR https://github.com/vmware-tanzu/tanzu-framework/pull/793
  a validation around this variable was added.
- Configuring this variable during addons upgrade as validation is
  failing the upgrade because of the missing config variable

### Which issue(s) this PR fixes

Fixes #1468

### Describe testing done for PR

- Added unit tests around configuring `TKG_HTTP_PROXY_ENABLED` during ugprade

### Release note

```release-note
Fix cluster upgrade failure error for proxied environment related to config variable `TKG_HTTP_PROXY_ENABLED` not set
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/tkg/client/upgrade_addon.go [hunk 1]
```diff
@@ -327,14 +327,17 @@ func (c *TkgClient) setProxyConfiguration(clusterClusterClient clusterclient.Cli
 
 	if httpProxy := configmap.Data["httpProxy"]; httpProxy != "" {
```
Comment (lubronzhan) [inline pkg/v1/tkg/client/upgrade_addon.go:328]: Does the configmap contains `TKG_HTTP_PROXY_ENABLED ` in it?

Comment (anujc25) [reply pkg/v1/tkg/client/upgrade_addon.go:328]: Unfortunately, it doesn't contain this information.

File: pkg/v1/tkg/client/upgrade_addon.go [hunk 2]
```diff
@@ -327,14 +327,17 @@ func (c *TkgClient) setProxyConfiguration(clusterClusterClient clusterclient.Cli
...
 	}
 
 	if httpsProxy := configmap.Data["httpsProxy"]; httpsProxy != "" {
 		c.TKGConfigReaderWriter().Set(constants.TKGHTTPSProxy, httpsProxy)
+		c.TKGConfigReaderWriter().Set(constants.TKGHTTPProxyEnabled, trueString)
 	}
 
 	if noProxy := configmap.Data["noProxy"]; noProxy != "" {
 		c.TKGConfigReaderWriter().Set(constants.TKGNoProxy, noProxy)
+		c.TKGConfigReaderWriter().Set(constants.TKGHTTPProxyEnabled, trueString)
```
Comment (gwang550) [inline pkg/v1/tkg/client/upgrade_addon.go:340]: I don't think you need to set `TKGHTTPProxyEnabled` when checking for noproxy. If httpproxy and httpsproxy doesn't exist, proxy is not enabled. Strickly speaking, httpproxy and httpsproxy have to exist at the same time, so checking only httpproxy is sufficient to determine whether `TKGHTTPProxyEnabled` need to be set or not

**Review by tenczar — APPROVED** [review]
LGTM

File: pkg/v1/tkg/client/upgrade_addon.go (hunk 2 above)
Comment (anujc25) [reply pkg/v1/tkg/client/upgrade_addon.go]: Thanks. Updated.

**Review by gwang550 — APPROVED** [review]
LGTM

**Review by lubronzhan — APPROVED** [review]

**Review by vuil — APPROVED** [review]
lgtm, thanks!

**Pull request merged**